source/postprocess/basic_statistics.cc

```diff
@@ -69,7 +69,11 @@ namespace aspect
               material_model.evaluate(in, out);
 
               const double gravity = this->get_gravity_model().gravity_vector(representative_point).norm();
-              const double Ra = material_model.reference_density()*
+
+              // check whether diffusivity is set to 0 (in case of backward advection)
+              const double Ra = (material_model.reference_thermal_diffusivity() == 0) ?
+                                std::numeric_limits<double>::infinity() :
+                                material_model.reference_density()*
```

Comment:
Not terribly important right now, but as a general remark for the future: this is a logically independent piece of the patch, having nothing much to do with the rest of the patch. As a general rule, we like it if independent patches are split into a separate pull request, since it makes reviewing (and, if necessary, undoing) a patch simpler.